### PR TITLE
Settings

### PR DESCRIPTION
**Pull request by daniel-sm-yu**
## Notion ticket link

[Settings Page](https://www.notion.so/uwblueprintexecs/Settings-Page-7b55ff46d7c04a26a1320ec5cdd28be5)
[Figma](https://www.figma.com/file/GhDd9C0XsyVDgwL5zzvDvO/HUG-S21?node-id=856%3A10017)

## Implementation description
* create the settings page with a table of admin users
* create 3 modals (ADD, EDIT, DELETE)
* create `DropDown` component with label and error message
* used `console.log` to mock sending requests to the back end
note: Chakra provides some defined sizes for Modals so the size doesn't exactly match the design

image
image
image
image
image
image

## Steps to test
1. run locally and go to `localhost:3000/admin/settings`
2. open chrome inspect
3. add, edit, delete users and see ensure it's reflected in the console
4. try to add or edit a user with a blank/invalid email, ensure an error appears

## What should reviewers focus on?
* the modals reflect the correct user

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/admin/settings.tsx
Comment (mkanoria) [inline pages/admin/settings.tsx:64]: Let's move all the table stuff to a different file for the component itself? Like `<SettingsTable>` and we can also add in the colums there and get data as a prop

File: pages/admin/settings.tsx
Comment (mkanoria) [inline pages/admin/settings.tsx:188]: Is the idea to call the `GET` endpoint again after the add action? Might as well just update the data locally

File: pages/admin/settings.tsx
Comment (mkanoria) [inline pages/admin/settings.tsx:205]: Same question here - we can discuss in more detail

**Review by mkanoria — COMMENTED** [review]
The PR looks good, great job on the Modals. Just a couple notes/questions. Should be good to merge after clarifying that
Tested out the UI locally and it looks great :D

File: pages/admin/settings.tsx
Comment (daniel-sm-yu) [reply pages/admin/settings.tsx:64]: 100% agree

File: pages/admin/settings.tsx
Comment (daniel-sm-yu) [reply pages/admin/settings.tsx:188]: discussed: will either return the updated user list from POST/DELETE requests OR have the front end call a GET request after every POST/DELETE

**Comment by mkanoria** [comment]
Looks great, thanks for making the changes! Would get another approval and then get this merged 🥳

**Review by mkanoria — APPROVED** [review]

**Review by yashgajaria — APPROVED** [review]
LGTM. Tested out the UI with the add, delete, and edit features and went through all the changes and all of it looks great!

**Pull request merged**
